File: pyg_sellergoods_interface/src/main/java/com/pyg/sellergoods/service/BrandService.java
package com.pyg.sellergoods.service;

import com.pyg.pojo.TbBrand;
import entity.PageResult;

import java.util.List;
import java.util.Map;

public interface BrandService {
    //    查找品牌
    public List<TbBrand> findAll();

    //  分页查找品牌
    public PageResult findPage(int pageNum, int pageSize);

    //添加品牌
    public void add(TbBrand brand);

    //修改品牌   第一实现回显 就是根据id查找品牌
    public TbBrand findOne(Long id);

    //第二步实现商品的修改
    public void update(TbBrand brand);

    //批量删除商品
    public void dele(Long[] ids);

    //查询操作
    public PageResult findPage(TbBrand brand,int pageNum, int pageSize);
    //查询品牌疯长的一个数组
    public  List<Map> selectOptionList();;
}

File: pyg_manager_web/src/main/webapp/js/service/brandService.js
//自定义一个服务
app.service("brandService",function ($http) {
    //查询全部的品牌
    this.findAll=function () {
        return $http.get("../brand/findAll.do");
    }
    //分页查询
    this.findPage=function (page, rows) {
        return $http.get("../brand/findPage.do?page=" + page + "&rows=" + rows);
    }
    //分页条件查询
    this.search=function (page,rows,searchEntity) {
        return $http.post("../brand/search.do?page=" + page + "&rows=" + rows,searchEntity);
    }
    //回显查询单个品牌
    this.findOne=function (id) {
        return $http.get("../brand/findOne.do?id=" + id);
    }
    //批量删除商品
    this.dele=function (seletIds) {
        return $http.get("../brand/dele.do?ids=" + seletIds)
    }
    //添加品牌
    this.add=function (entity) {
        return $http.post("../brand/add.do",entity);
    }
    //修改品牌
    this.update=function (entity) {
        return $http.post("../brand/update.do",entity);
    }

    this.selectOptionList=function () {
        return $http.get("../brand/selectOptionList.do");
    }
})